src/models/__tests__/profile.test.ts

```diff
@@ -45,40 +45,66 @@ const aRetrievedProfile: RetrievedProfile = {
   ...aStoredProfile
 };
 
+const aLastAppVersion = "1.0.0";
+
 describe("findLastVersionByModelId", () => {
-  it("should resolve to an existing profile", async () => {
-    const containerMock = ({
-      items: {
-        create: jest.fn(),
-        query: jest.fn(() => ({
-          fetchAll: jest.fn(() =>
-            Promise.resolve({
-              resources: [aRetrievedProfile]
-            })
-          )
-        }))
+  it.each`
```

Comment:
```suggestion
  it.each`
    case                                         | lastAppVersion     | expectedLastAppVersion | reminderStatus | expectedReminderStatus
    ${"existing profile"}                        | ${undefined}       | ${"UNKNOWN"}           | ${undefined}   | ${"UNSET"}
    ${"existing profile with last app version"}  | ${aLastAppVersion} | ${aLastAppVersion}     | ${null}        | ${"UNSET"}
    ${"existing profile with reminder disabled"} | ${undefined}       | ${"UNKNOWN"}           | ${"DISABLED"}  | ${"DISABLED"}
    ${"existing profile with all valid props"}   | ${aLastAppVersion} | ${aLastAppVersion}     | ${"ENABLED"}   | ${"ENABLED"}
  `(
```

some case is redundant: e.g. we are testing status `ENABLED` two times and we are not testing null case. I think the suggestion above can be an improvement

Reply:
we are testing it against a different initial state, and null (ditto) is not coherent at this stage.

Reply:
removing null testing, we are currently testing `ENABLED` two times so i think we can remove one to get this test simpler
```suggestion
  it.each`
    case                                         | lastAppVersion     | expectedLastAppVersion | reminderStatus | expectedReminderStatus
    ${"existing profile"}                        | ${undefined}       | ${"UNKNOWN"}           | ${undefined}   | ${"UNSET"}
    ${"existing profile with reminder disabled"} | ${undefined}       | ${"UNKNOWN"}           | ${"DISABLED"}  | ${"DISABLED"}
    ${"existing profile with all valid props"}   | ${aLastAppVersion} | ${aLastAppVersion}     | ${"ENABLED"}   | ${"ENABLED"}
  `(
```

Reply:
This test is testing the decoders are working properly when we retrieve a profile from cosmos.
We are testing that profile decoder is returning the right properties with all values, when no lastAppVersion is present, when no reminderStatus is present, when we have of the two and both.

Reply:
I understand what you are saying. what I meant to say was that the two variables are not dependant to eachother so i think another `it.each` could have been done just for `reminderStatus`, leaving the `it.each` for the `lastAppVersion` alone.